addons/mail/static/src/models/composer/composer.js

```diff
@@ -272,99 +272,177 @@ function factory(dependencies) {
             await this.env.bus.trigger('do-action', { action, options });
         }
 
+        /**
+         * Recursive function that process the queue.
+         */
+        async processMessagesToBeSent() {
```

Comment:
> ("process" is synonym to "do" in the domain of programming).

I don't see it that way.
I expect a `DoSomething` function to start and end inside the function block (and maybe return a result).
I expect a `ProcessSomething` function to start some treatment that will run for a period of time.

`DoOpenFile` => I expect the function to return the file content.
`ProcessFileToExport` => Will do something with the file and call some export function.

> I think you should have different interfaces. At component level, it's used to send again, so the dispatch to model from component should be named something like "sendAgain" to this message, not the composer. It's not correct functionally that clicking try again on a single message try again message post on all pending messages.

I don't follow here. The retry button just start the processing of the message, it's not a sendAgain, it a restart the processing. It's done, regardless of the message it self.

Reply:
> I expect a `DoSomething` function to start and end inside the function block (and maybe return a result).
> I expect a `ProcessSomething` function to start some treatment that will run for a period of time.

Both are too technical for a public API. Also in both cases, if it's fine to name function that way, you need comment to clarify what is expected by "do" and "process", because both verbs are fuzzy by themselves. Usually such named functions should be private.

> I don't follow here. The retry button just start the processing of the message, it's not a sendAgain, it a restart the processing. It's done, regardless of the message it self.

You think too much of how everything works technically, so much that it's leaked to the public API. It shouldn't: when user clicks on "send again", it should _simply_ send the message again. "process messages to be sent" is far too technical and implementation-detail dependent. It's okay for the inner guts of model to actually "process messages to be sent", but the public API should NOT leak that, hence my suggestion to define different interfaces (one public with name at functional level, and the other private with maybe current naming).

Reply:
So, *in concreto*, you want:
```javascript
postMessage() {
     return this._processMessageToBeSent();
}

_processMessageToBeSent() {
    /* ... */
}
```
I don't really see the point of having this alias. What do I win beside getting angry because i have to jump one more function to find the real stuff ?

Reply:
I don't like the new functions like `processMessageToBeSent` in this PR, for several reasons (amount of LOC, name of function, recursive aspect mixed with public/private/recursion usage, etc.).

The proper function name between component and model is a step in the right direction. The necessary changes on `processMessageToBeSent` will follow after this change and the changes you could made from my comments as of now, in addition to new waves of comment to iterate only we are all happy